src/plugins/blocksmith/components/divider-block-modal/index.jsx

```diff
@@ -0,0 +1,198 @@
+    this.handleChangeStrokeColor = this.handleChangeStrokeColor.bind(this);
+    this.handleChangeStrokeStyle = this.handleChangeStrokeStyle.bind(this);
+    this.handleEditBlock = this.handleEditBlock.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeTime = this.handleChangeTime.bind(this);
+    this.handleChangeStrokeColor = this.handleChangeStrokeColor.bind(this);
+  }
+
+  componentDidMount() {
+    this.textElement.focus();
+  }
+
+  setBlock() {
+    const { hasUpdatedDate, text, strokeColor, strokeStyle, updatedDate } = this.state;
+    const { block } = this.props;
+    const setBlock = block.schema().createMessage()
+      .set('text', text)
+      .set('stroke_style', strokeStyle)
+      .set('updated_date', hasUpdatedDate ? updatedDate.toDate() : null);
+
+    if (strokeColor) {
+      setBlock.set('stroke_color', strokeColor.replace(/\s/g, '-').replace(/[A-Z]/g, (m) => m.toLowerCase()));
+    }
+
+    return setBlock;
+  }
+
+  handleAddBlock() {
+    const { onAddBlock, toggle } = this.props;
+    onAddBlock(this.setBlock());
+    toggle();
+  }
+
+  handleEditBlock() {
+    const { onEditBlock, toggle } = this.props;
+    onEditBlock(this.setBlock());
+    toggle();
+  }
+
+  handleChangeDate(date) {
+    this.setState(changedDate(date));
+  }
+
+  handleChangeTime({ target: { value: time } }) {
+    this.setState(changedTime(time));
+  }
+
+  handleChangeHasUpdatedDate() {
+    this.setState(({ hasUpdatedDate }) => ({ hasUpdatedDate: !hasUpdatedDate }));
+  }
+
+  handleChangeInput({ target: { id, value } }) {
+    this.setState({ [id]: value });
+  }
+
+  handleChangeStrokeColor({ value: strokeColor }) {
+    this.setState({ strokeColor });
+  }
+
+  handleChangeStrokeStyle({ value: strokeStyle }) {
+    this.setState({ strokeStyle });
+  }
+
+  render() {
+    const {
+      hasUpdatedDate,
+      isValid,
+      text,
+      strokeColor,
+      strokeStyle,
+      updatedDate,
+    } = this.state;
+    const { isFreshBlock, isOpen, toggle } = this.props;
+
+    return (
+      <Modal isOpen={isOpen} toggle={toggle}>
+        <ModalHeader toggle={toggle}>
+          {`${isFreshBlock ? 'Add' : 'Update'} Divider Block`}
+        </ModalHeader>
+        <ModalBody className="p-0">
+          <FormGroup className="px-4">
+            <Label className="pt-4">Text</Label>
+            <Input
+              className="mb-3"
+              id="text"
+              innerRef={(el) => { this.textElement = el; }}
+              name="content"
+              onChange={this.handleChangeInput}
+              placeholder="Enter heading text here"
+              type="textarea"
+              value={text || ''}
+              required={false}
+            />
+            <PicklistPicker label="Stroke Color" name="strokeColor" picklistId="divider-block-colors" onChange={this.handleChangeStrokeColor} value={{ label: strokeColor, value: strokeColor }} isEditMode />
```

Comment:
@cohencodes break this onto multiple lines

also, the convention for picklist names is `${item}-${pluralize(fieldName)}` so this should be `divider-block-stroke-colors`